server/app/lib/actions/fusor/deployment/rhev/ose_launch.rb

```diff
@@ -139,6 +153,96 @@ def wait_for_sshd_on_host(host, username, password)
             ::Fusor.log.error "====== waiting for ssh on #{host} TIMED OUT! ======"
             return false
           end
+
+          def find_hostgroup(deployment, name)
+            # locate the top-level hostgroup for the deployment...
+            # currently, we'll create a hostgroup with the same name as the
+            # deployment...
+            # Note: you need to scope the query to organization
+            parent = ::Hostgroup.where(:name => deployment.label).
+                joins(:organizations).
+                where("taxonomies.id in (?)", [deployment.organization.id]).first
+
+            # generate the ancestry, so that we can locate the hostgroups
+            # based on the hostgroup hierarchy, which assumes:
+            #  "Fusor Base"/"My Deployment"
+            # Note: there may be a better way in foreman to locate the hostgroup
+            if parent
+              if parent.ancestry
+                ancestry = [parent.ancestry, parent.id.to_s].join('/')
+              else
+                ancestry = parent.id.to_s
+              end
+            end
+
+            # locate the engine hostgroup...
+            ::Hostgroup.where(:name => name).
+                where(:ancestry => ancestry).
+                joins(:organizations).
+                where("taxonomies.id in (?)", [deployment.organization.id]).first
+          end
```

Comment:
find_hostgroup looks like a copied method :) I've seen this else where in the code. I'll accept it for now, we'll probably have to clean this up one day. HA! yep it lives in ```trigger_provisioning.rb``` :)